### ISSUE TITLE
`isNullOrBlank` can not be used with nullable strings

### ISSUE THREAD

**Issue by JLLeitschuh** [npryce/hamkrest#44]
The following test does not compile but should:
```kotlin
val message: String? = null
assertThat(message, isNullOrBlank)
```

**Comment by SirWellington** [comment]
I agree. I had to make my own `Matcher`: 

```kotlin
val isNull: Matcher<Any?> = Matcher(Objects::isNull)

val isNullOrEmptyString: Matcher<CharSequence?> = isNull.or(isEmptyString as Matcher<CharSequence?>)
```

And use it as such:

```kotlin
assertThat(message, isNullOrEmptyString)
assertThat(message, !isNullOrEmptyString)
```

**Comment by npryce** [comment]
Fixed by #43
